### ISSUE TITLE
Extreme lag for players when this module is enabled

### ISSUE THREAD

**Issue by rockwallguy** [CDeenen/SharedVision#8]
I haven't tested this on 1.0.6, but the update notes didn't mention it.  It's hard for me to test because it only affects my players, but I had two separate groups complaining of massive lag.  They were complaining of delays of up to a second between the time they'd move the token and when it would move.  It was exacerbated in narrow caverns with lots of wall segments.  The lag persisted whether the global shared vision toggle was on or off.  

I had two different groups complain of this issue. It cropped up after I updated to 8.6 about a week after release.  I always stay current on module updates.  Last night (Sunday the 20th, the day before you released 1.0.6, I ran the Find the Culprit mod while my players were in game to help me test, and they identified it as Shared Vision.  After disabling Shared Vision, the rest of the session went completely smooth.  

I could not replicate this myself.  Logging in as a player in a separate browser window from the server (I self-host) was not able to recreate the issue.  I'm not sure if that means it's related to having a separate IP or if it breaks when multiple players are on?  I'm not sure what to say about that.

Even as I type this, I know this is a weird one to track down, but I felt the need to mention it because it was completely game-breaking.  I'll try testing at my next game on Friday and see if the 1.0.6 update has corrected it.

**Comment by CDeenen** [comment]
Shared Vision does some extra calculations whenever the lighting is updated, however, this is the first time that I get a report like this. It is indeed a bit difficult to track down, but I'll give it a go. I have a game later this week and see if my players notice any difference when I disable the module.

**Comment by GambetTV** [comment]
Yeah, I just came to report this exact thing. Any map I have Vision Enabled on suffers horribly from lag for me and all of my players. At first I thought it was light sources too but it doesn't seem to be. It might be exacerbated by light sources and complicated walls, but even simple walls cause a lot of issues even if this is the only module enabled.

You might not notice it as bad if you drag your tokens instead of WASD them around the map. But try WASDing and watch how the shadows bounce around like crazy as the token moves, and how janky the movement becomes.

I should note that as the GM I do not experience this that much, but it's nearly unplayable for my players, even if their tokens aren't sharing vision. I tested this both in Chrome and Firefox and it happens in both browsers, but not from the main client.

I hope this is fixable, I've been using this module for a couple months now and it's excellent, but maybe after the last update or two something went wrong? I'm not sure, I didn't notice it until Monday.

BTW this problem exists in both Foundry 0.7.10 and 0.8.8.

**Comment by CDeenen** [comment]
This week's game was unfortunately cancelled, so I wasn't able to test it with my players.
I tried try to reproduce it on my own on both 0.7.10 and 0.8.8, and couldn't notice any difference between having Shared Vision enabled or not.
I'll report again after the next session.

**Comment by CDeenen** [comment]
I did some testing during last night's game, with 3 players connected. After disabling the module, 1 didn't notice any difference, 1 noticed a very slight performance increase (she was on a slower computer), and 1 actually noticed a slight perfomance decrease (I don't know how, though).
Right now, I'm not sure how to proceed with this issue.

**Comment by HLennart** [comment]
Hey I've just experienced the same problem last night. Player experience was on a spectrum from 1sec lag to 15-20seconds of lag. When i subsequently disabled this module (and only this module), performance went back to normal.

For what it's worth, I as the host didn't experience any lag, no matter if I joined via the desktop application or a browser.

If you want to try it out, get back to me and we can arrange for your to join my foundry session.

**Comment by Kingcrab-cpu** [comment]
Hi,

We played pf 2 extinction curse last night on hermitage map remake (https://foundryvtt.com/packages/ec-maps-remake-by-lios) with shared vision enabled and it was causing huge lag for all player ( 10sec to complete freeze). We only identified the problem pretty late , but disabling the module fixed it. Seems it s only for this map , i tested on other maps without lag.

**Comment by EvilRenegade** [comment]
I just accidentally came across this issue while checking if SharedVision was v9 compatible; it struck me because over the past two games, I've noticed massive lag on the player side.
I could see it rather clearly because I had two sessions open on the same computer - one for the DM, behaving perfectly fine, and one for the players, shared via Discord.

I noticed repeatedly that the player view was half a minute or more behind, and at times, I had to interrupt my descriptions and watch the tokens crawl to where I dragged them, because my players just weren't up to speed yet.

I don't know if SharedVision was the culprit, but I did check the performance graphs when it happened and the machine itself was fine.

The extension is deactivated right now because the v9 upgrade notes said to deactivate all extensions related to view (hence my visit).
I'll leave it off and see how the next game goes.

**Comment by AlexeSimon** [comment]
Same issue here for my 4 players. v9

**Comment by CDeenen** [comment]
I really don't know what to do about this issue. I have not been able to reproduce it and this doesn't appear to affect that many people. From the reports I cannot really narrow the search down to anything.
I am planning an update soon, so I'll keep my eye out for any code that might be the culprit. That's basically all I can do at this point.
